src/rgw/rgw_sal_motr.cc

```diff
@@ -2649,6 +2664,89 @@ int MotrAtomicWriter::process(bufferlist&& data, uint64_t offset)
   return this->write();
 }
 
+int MotrObject::update_null_version_index(const DoutPrefixProvider *dpp, rgw_bucket_dir_entry& ent)
+{
+  int rc, ret_code = -ENOENT;
+  string tenant_bkt_name = get_bucket_name(this->get_bucket()->get_tenant(), this->get_bucket()->get_name());
+
+  string bucket_index_iname = "motr.rgw.bucket.index." + tenant_bkt_name;
+  std::string null_version_key = this->get_name() + "/null";
+  bufferlist::const_iterator iter;
+  bufferlist bl, bl_null_idx_val;
+  rgw_bucket_dir_entry current_null_key_ref;
+  current_null_key_ref.key.name = this->get_name();
+  current_null_key_ref.key.instance = this->get_instance();
+  current_null_key_ref.encode(bl_null_idx_val);
```

Comment:
instead of using `rgw_bucket_dir_entry` to store null object reference, can't just store the index entry of null object (i.e. `object_key[versionid]`) directly.
We should write separate functions for encode and decode of null object reference which will just encode or decode the index entry of null object.

Reply:
We have found `cls_rgw_obj_key` structure which stores key and instance only, it also has encode() and decode() method defined in cls_rgw_types.h and cls_rgw_types.cc files. 
We have made code changes, but facing few issues with some test cases if we use `cls_rgw_obj_key` instead of `rgw_bucket_dir_ent`.